frappe/model/db_query.py

```diff
@@ -127,6 +127,22 @@ def build_and_run(self):
 		if self.distinct:
 			args.fields = 'distinct ' + args.fields
 			args.order_by = '' # TODO: recheck for alternative
+			
+		# Postgres requires any field that appears in the select clause to also 
+		# appear in the order by and group by clause
+		if args.order_by and args.group_by:
+			order_field = args.order_by
+
+			for r in (" order by ", " asc", " ASC", " desc", " DESC"):
+				order_field = order_field.replace(r, "")
+
+			if order_field not in args.fields:
+				order_fieldm = order_field.replace("`", "")
+				if "." in order_fieldm:
+					args.fields += ", MAX(" + order_fieldm.split(".")[1] + ") as `" + order_fieldm + "`"
+				else:
+					args.fields += ", MAX(" + order_fieldm + ") as `" + order_fieldm + "`"
```

Comment:
```suggestion
				order_fieldm = order_field.replace("`", '"')
				args.fields += f", MAX({order_fieldm}) as `{order_fieldm}`"
```

We don't need to separate the column from the table so I think this works too. Although I don't wholly understand the reason for adding `MAX(...)` here.

---

Also, I'd imagine we'd do some post-processing to remove the extra fields. This step seems unnecessary for results of type `List[Dict]` but could be problematic _(& possibly breaking)_ while `as_list=True`.

Reply:
So I already incorporated the changes here that you suggested (it just doesn't show up here because they are combined with another change). Do we need to include said post-processing to remove the extra fields as part of this PR? If so, where should I include that?

Reply:
After testing the more concise format I discovered that it was necessary to actually separate the column to the table. The reason being is because `tabTable.field` needs to be converted `tabTable`.`field` for this to work properly so we do need to parse the string. 

Also, I noticed that I did not answer your original question regarding MAX(). The  reason the MAX() is necessary is because in some circumstances frappe inserts docstatus and modified fields into the results of sql calls even if you did not specificially request them. That combined with the requirement of aggregate function for postgres means that those fields must be included as an aggregate function. MAX was chosen here because it just gives the most recent/highest of these fields. An example of where this happens upstream is the function get_dashboard_info